### DISCUSSION TITLE
Weighted Burger Vector

### DISCUSSION THREAD

**Discussion by Belt91** [mtex-toolbox/mtex#1930] [Ask Anything]
Hi,

I am trying to explore the recently added calculation for the Weighted Burger Vectors in MTeX 5.10 . I saw on the documentation ( which btw could be improved with a bit of theory and a touch of what it represents) that the calculation is accomplished by a simple function call, but when I run it I get the following error :

![image](https://github.com/mtex-toolbox/mtex/assets/134724391/e55f8b4a-4162-4d4b-a35c-9aa6b939305a)

the ebsdS data I use as input are simply ebsdS data after data cleaning but I think the error does not depend on it. Any help il welcome.

Best,

Marco

**Comment by ralfHielscher** [comment]
Hi Marco,

indeed we are lacking a good documentation page for the Burgers vector. Maybe you can help :) If you have a nice data set and some practical question that could be discussed using the Burgers vector I would be glad if you could set up the docu page together. In the mean time you need to to first gridify you data, i.e.,
```matlab
ebsd = ebsd.gridify
W = ebsd.weightedBurgersVector
```
All the best,
Ralf.

**Reply by Belt91** [reply]
Hi Ralf,

thank you very much for the quick response. Now it doeas the calculation (please add this line of code to the documentation :) ) but the output is rather difficult to interpret. I just found on a recent Application note from Oxford Instruments  ( available after a free subscription at the following [website](https://nano.oxinst.com/campaigns/downloads/evaluating-dislocation-densities-and-slip-systems-in-deformed-titanium-using-ebsd?utm_campaign=NanoAnalysis+&utm_content=LinkedIn&utm_medium=social&utm_source=LinkedIn&19_form_success=1)  a nice representation and use of WBV to identify slip systems and dislocation kinds. It looks like this:

![image](https://github.com/mtex-toolbox/mtex/assets/134724391/91305c4a-12be-4a64-8a85-a73223f4bc65)

Is there already a way to obtain this kind of plots or could it be implemented?

Best,

Marco

**Reply by ralfHielscher** [reply]
Yes this should be possible by
```matlab
ebsd = ebsd.gridify
W = ebsd.weightedBurgersVector;

% turn into crystal coordinates
WC = inv(ebsd.orientations .* W);

% perform density estimation 
WDistri = calcDensity(WC);

plot(WDistri)
```
Most likely you need to do some filtering on the Burgers vectors first. Maybe remove those at the grain boundaries ....

Ralf.

**Reply by Belt91** [reply]
Ralf, 

thank you for your feedback.  Unfortunately it is not over yet!

I implemented the code you provided but I am given a dimension-compatibility error when computing the inverse matrix:

![image](https://github.com/mtex-toolbox/mtex/assets/134724391/ca6955bc-89ab-4b48-93a1-4707b71e4f13)

Besides, how can I filter the vectors at the boundaries? I checked among the "." methods but I cannot figure it out.

Thanks,

Marco

**Reply by Belt91** [reply]
I think I identified the error by comparing with the documentation page. The steps are
```matlab
ebsdSGrid=ebsdS.gridify                        % Prepare the EBSD data 
W=weightedBurgersVec(ebsdSGrid)      % Compute the Weighted Burger Vectors (WBV)
WC= inv(ebsdSGrid.orientations).* W    % Transforms WBV into crystal coordinates
WDistri = calcDensity(WC);                     % Compute the density of WC

plot(WDistri)
mtexColorMap('hot)
mtexColorBar
```
Now the plot is like this, which is close to what I was looking for.

![image](https://github.com/mtex-toolbox/mtex/assets/134724391/484fbb57-ab15-4080-9262-6a2bf234e45e)

I am doubtful because of the very short range of the density distribution. The three directions here are crystallographic planes normal or directions?

Thank you for your help.


Best,

 Marco

**Reply by ralfHielscher** [reply]
You might want to include the norm of the weighted Burgers vectors as weight for the density calculation. As it is now, each vector is treated with the same weight. However, for Burger vectors with very small length the axis is not very well defined. You can include weights in the following way:
```matlab
ebsdSGrid=ebsdS.gridify                        % Prepare the EBSD data 
W=weightedBurgersVec(ebsdSGrid)      % Compute the Weighted Burger Vectors (WBV)
WC= inv(ebsdSGrid.orientations).* W    % Transforms WBV into crystal coordinates
WDistri = calcDensity(WC,'weights',norm(W));      % Compute weighted density of WC

plot(WDistri)
mtexColorMap('hot)
mtexColorBar
```
Ralf.

**Reply by DriptD** [reply]
Hi @ralfHielscher,

I tried executing the above code. But encountered the following error at the line that computes the weighted density of WC. Could you please help?

```
Arrays have incompatible sizes for this operation.

Error in S2FunHarmonic.quadrature (line 90)
  nfsftmex('set_f', plan, W(:) .* values(:, index));

Error in Miller/calcDensity (line 36)
sF = 4*pi * S2FunHarmonic.quadrature(h,ones(size(h)),varargin{:});
```
I am using MTEX 5.10.0 in Matlab 2021a. Thanks!

Sincerely,
Dripta
